src/main/java/pro/sky/Coursework2/service/JavaQuestionService.java

```diff
@@ -0,0 +1,52 @@
+package pro.sky.Coursework2.service;
+
+import pro.sky.Coursework2.exeption.QuestionAlreadyAddedException;
+import pro.sky.Coursework2.exeption.QuestionNotFoundExeption;
+import pro.sky.Coursework2.model.Question;
+import org.springframework.stereotype.Service;
+
+import java.util.*;
+
+@Service
+public class JavaQuestionService implements QuestionService {
+    private final Set<Question> questions = new HashSet<>();
+    private final Random random = new Random();
+
+    @Override
+    public Question add(String question, String answer) {
+        return add(new Question(question, answer));
+    }
+
+    @Override
+    public Question add(Question question) {
+        for (Question q : questions) {
+            if (q.equals(question)) {
+                throw new QuestionAlreadyAddedException();
+            }
+        }
+        questions.add(question);
```

Comment:
```suggestion
        if (!questions.add(question)) {
            throw new QuestionAlreadyAddedException();
        }
```
код значительно упрощается, если правильно использовать методы коллекций